### PR TITLE
Clu

### PR DESCRIPTION
**Pull request by BurtBiel**
This PR now contains 2 changes from today, 1) updating the default format to not include object-based ToStrings, and 2) the rename of the cmdlet cli usage params

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@BurtBiel__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (burtbiel). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by markcowl** [comment]
@azuresdkci retest this please

File: examples/compute-management/02-VirtualMachineCreation.sh [hunk 1]
```diff
@@ -3,15 +3,15 @@ set -e
 printf "\n=== Managing Virtual Machine Creation in Azure Compute ===\n"
 
 printf "\n1. Creating a new resource group: %s and location: %s.\n" "$groupName" "$location"
-az group create -n "$groupName" --location "$location"
+az resourcemanager group create -n "$groupName" --location "$location"
```
Comment (markcowl) [inline examples/compute-management/02-VirtualMachineCreation.sh]: This is the most used cmdlet in the entire set opf cmdlets - you have to create a resource group to do anything else.  Do we really want the cmdlet name to be so wordy?

File: examples/compute-management/02-VirtualMachineCreation.sh [hunk 2]
```diff
@@ -3,15 +3,15 @@ set -e
...
 
 printf "\n2. Creating a new storage account '%s' in type '%s'.\n" "$storageAccountName" "$storageAccountType"
 az storage account new --resourcegroupname "$groupName" --name "$storageAccountName" --location "$location" --type "$storageAccountType"
 
 printf "\n3. Create virtual network.\n"
-result=`az virtual network new --resourcegroupname "$groupName" --name test --location "$location" --addressprefix "[\"10.0.0.0/16\"]" --subnet "[{\"Name\":\"test\",\"AddressPrefix\":\"10.0.0.0/24\"}]" --force`
+result=`az network vnet new --resourcegroupname "$groupName" --name test --location "$location" --addressprefix "[\"10.0.0.0/16\"]" --subnet "[{\"Name\":\"test\",\"AddressPrefix\":\"10.0.0.0/24\"}]" --force`
```
Comment (markcowl) [inline examples/compute-management/02-VirtualMachineCreation.sh]: Are we realy going to confuse network vnets with other kinds of vnets?  I would think vnet would be a top level noun

File: examples/resource-management/02-Resource.sh [hunk 3]
```diff
@@ -3,44 +3,44 @@ set -e
...
 
 printf "\n1. Creating a new resource group: %s and location: %s.\n" "$groupName" "$location"
-az group create --name "$groupName" --location "$location"
+az resourcemanager group create --name "$groupName" --location "$location"
 destinationGroupName=$groupName"Destination"
 
 printf "\n2. Registering Resource Provider Namespace.\n"
 providerNamespace="Providers.Test"
-az resource provider register --ProviderNamespace $providerNamespace --Force
+az resourcemanager resource resourcemanager provider register --ProviderNamespace $providerNamespace --Force
```
Comment (markcowl) [inline examples/resource-management/02-Resource.sh]: Again, seems like 'resourcemanager' is wordy and unnecessary to disambiguate

File: examples/resource-management/03-Deployments.sh [hunk 4]
```diff
@@ -3,14 +3,14 @@ set -e
...
 
 printf "\n3. Test template with JSON parameter object\n"
-az group deployment test -g "$groupName" --templatefile $BASEDIR/sampleTemplate.json --templateparameterobject "{\"siteName\":\"$resourceName\",\"hostingPlanName\":\"$resourceName\",\"siteLocation\":\"$location\",\"workerSize\": 0 }"
+az resourcemanager group deployment test -g "$groupName" --templatefile $BASEDIR/sampleTemplate.json --templateparameterobject "{\"siteName\":\"$resourceName\",\"hostingPlanName\":\"$resourceName\",\"siteLocation\":\"$location\",\"workerSize\": 0 }"
 
 printf "\n4. Provisioning Deployment\n"
-deploymentInfo=`az group deployment create --Name "$resourceName" --ResourceGroupName "$groupName" --TemplateFile $BASEDIR/sampleTemplate.json --TemplateParameterFile $BASEDIR/sampleTemplateParams.json`
+deploymentInfo=`az resourcemanager group deployment create --Name "$resourceName" --ResourceGroupName "$groupName" --TemplateFile $BASEDIR/sampleTemplate.json --TemplateParameterFile $BASEDIR/sampleTemplateParams.json`
```
Comment (markcowl) [inline examples/resource-management/03-Deployments.sh]: This is certainly in the top few commands used by everyone and this name is just too wordy.  Why not az deployment create?

File: examples/resource-management/04-RoleAssignments.sh [hunk 5]
```diff
@@ -3,25 +3,25 @@ set -e
...
+az resourcemanager group create --name "$groupName" --location "$location"
 
 printf "\n2. Creating a new Role Assignment.\n"
 export MSYS_NO_PATHCONV=1
-users=$(az ad users get)
+users=$(az activedirectory users ls)
 userId=$(echo $users | cat | jq '.[0].Id' -s --raw-output)
 echo "UserID: $userId"
-roleDefinitions=$(az role definition get)
+roleDefinitions=$(az security role definition ls)
```
Comment (markcowl) [inline examples/resource-management/04-RoleAssignments.sh]: Same comment here - the security noun seems ambiguous

az role definition create
az role assignment create

seem more natural

File: examples/webapp-management/01-AppServicePlan.sh [hunk 6]
```diff
@@ -49,21 +49,21 @@ whpInfo2=`az app service plan get --name $whpName2`
...
 [  $plansInSubscription == *"$whpName"* ]
 [  $plansInSubscription == *"$whpName2"* ]
 
 printf "\n9. Remove app service plan: %s." "$whpName"
-az app service plan remove -n $whpName -g $groupName
+az appservice plan rm -n $whpName -g $groupName
 
 printf "\n10. Remove app service plan: %s." "$whpName2"
-az app service plan remove -n $whpName2 -g $groupName
+az appservice plan rm -n $whpName2 -g $groupName
```
Comment (markcowl) [inline examples/webapp-management/01-AppServicePlan.sh]: What is the distinction between appservice and webapp?  We should decide on one noun for this and use it throughout:

az appservice create
az appservice plan create
az appservice slot create

etc.

File: examples/compute-management/02-VirtualMachineCreation.sh (hunk 1 above)
Comment (BurtBiel) [inline examples/compute-management/02-VirtualMachineCreation.sh]: Joe/Jason would like to keep the resourcemanager term.  Is it possible to create an alias for the term?

File: examples/compute-management/02-VirtualMachineCreation.sh (hunk 2 above)
Comment (BurtBiel) [inline examples/compute-management/02-VirtualMachineCreation.sh]: Changed to top-level noun

File: examples/resource-management/02-Resource.sh (hunk 3 above)
Comment (BurtBiel) [inline examples/resource-management/02-Resource.sh]: Joe or Jason may be able to comment on this further

File: examples/resource-management/03-Deployments.sh (hunk 4 above)
Comment (BurtBiel) [inline examples/resource-management/03-Deployments.sh]: That seems like it might be good wording, can you run it by Joe?

File: examples/resource-management/04-RoleAssignments.sh (hunk 5 above)
Comment (BurtBiel) [inline examples/resource-management/04-RoleAssignments.sh]: Jason voted for keeping the security term to keep azure security concepts in a single location and be extra clear about what the role was

File: examples/webapp-management/01-AppServicePlan.sh (hunk 6 above)
Comment (BurtBiel) [inline examples/webapp-management/01-AppServicePlan.sh]: Updated to app service instead of webapp

Comment (JasonRShaver) [inline examples/webapp-management/01-AppServicePlan.sh]: appservice (one word) right?

Comment (mohitsriv) [inline examples/webapp-management/01-AppServicePlan.sh]: The branding name is in fact two words: https://azure.microsoft.com/en-us/services/app-service/. 

 

Mohit

File: examples/resource-management/03-Deployments.sh (hunk 4 above)
Comment (JasonRShaver) [inline examples/resource-management/03-Deployments.sh]: If we used 'deployment', we would run into confusing names with the deployment records: "azure deployment group deployment list"

Comment (JasonRShaver) [inline examples/resource-management/03-Deployments.sh]: Are you thinking of promoting the Azure Group Deployment subcommand to be its own command?

File: examples/webapp-management/01-AppServicePlan.sh (hunk 6 above)
Comment (mohitsriv) [inline examples/webapp-management/01-AppServicePlan.sh]: The branding is in fact two words: https://azure.microsoft.com/en-us/services/app-service/.  But, one word is fine for the command itself.

File: examples/compute-management/02-VirtualMachineCreation.sh (hunk 1 above)
Comment (markcowl) [inline examples/compute-management/02-VirtualMachineCreation.sh]: @BurtBiel  The decision we came to this morning was to use azure resourceGroup [verb]

File: examples/compute-management/02-VirtualMachineCreation.sh (hunk 2 above)
Comment (markcowl) [inline examples/compute-management/02-VirtualMachineCreation.sh]: @BurtBiel  It looks like the scenario test script did not change, we should change it as part of this PR.  Can I ask you to run the scenario tests locally?  We can help with doing the necessary setup, but most of it is here:  https://github.com/Azure/azure-powershell/blob/clu/clu-getstart.md#testing-cmdlets

File: examples/resource-management/02-Resource.sh (hunk 3 above)
Comment (markcowl) [inline examples/resource-management/02-Resource.sh]: @BurtBiel so this is clearly incorrect - even if we keep resourceManager, it should not appear twice.

File: examples/resource-management/03-Deployments.sh (hunk 4 above)
Comment (markcowl) [inline examples/resource-management/03-Deployments.sh]: @BurtBiel @JasonRShaver The consensus was azure resourceGroup deployment create.

**Comment by markcowl** [comment]
@BurtBiel There are merge conflicts, please update, and please ensure that scenario tests pass, as it looks like at least some shell example scripts were not updated.

**Comment by BurtBiel** [comment]
I'm getting setup to check the scenario tests now, I'll let you know the results

File: src/CLU/Commands.ResourceManager.Cmdlets/Implementation/GetAzureResourceGroupDeploymentOperationCmdlet.cs [hunk 7]
```diff
@@ -26,6 +26,7 @@ namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation
     /// Gets the deployment operation.
     /// </summary>
     [Cmdlet(VerbsCommon.Get, "AzureRmResourceGroupDeploymentOperation"), OutputType(typeof(PSResourceObject))]
+    [CliCommandAlias("resourcemanager;group;deployment;operation;ls")]
```
Comment (stankovski) [inline src/CLU/Commands.ResourceManager.Cmdlets/Implementation/GetAzureResourceGroupDeploymentOperationCmdlet.cs]: Why don't we use "space" instead of semicolon for noun separator

**Comment by BurtBiel** [comment]
about half the tests passed, the failures appear to be for reasons other than the rename.  


. 02-Resource.sh
ERROR: The value passed for ResourceId must not be null

. 04-RoleAssignments.sh
2. Creating a new Role Assignment.
Exception of type 'Microsoft.Rest.Azure.CloudException' was thrown.
ERROR: Exception of type 'Microsoft.Rest.Azure.CloudException' was thrown.

. 05-RoleDefinitions.sh
ERROR: The client 'burtbiel@microsoft.com' with object id '03f1f1bb-c826-4c7b-b368-940c2aa242df' does not have authorization to perform action 'Microsoft.Authorization/roleDefinitions/write' over scope '/subscriptions/2c224e7e-3ef5-431d-a57b-e71f4662e3a6'.

. 01-Storage-Accounts.sh 
Already broken, not updated to use az instead of azure

. 01-VirtualHardDisks.sh
test hung

.  01-AppServicePlan.sh - 03-WebApp
All appear to be busted, syntax errors
1. Create a new app service plan  The value passed for Name must not be empty
ERROR: The value passed for Name must not be empty
